kotlin/change-basemaps/src/main/java/com/esri/arcgisruntime/sample/switchbasemaps/MainActivity.kt

```diff
@@ -0,0 +1,167 @@
+/* Copyright 2018 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.switchbasemaps
+
+import android.content.res.Configuration
+import android.os.Bundle
+import android.support.v7.app.ActionBarDrawerToggle
+import android.support.v7.app.AppCompatActivity
+import android.view.MenuItem
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  private var mNavigationDrawerItemTitles: Array<String>? = null
+
+  private var mDrawerToggle: ActionBarDrawerToggle? = null
```

Comment:
For this one, you want it to be a `val` and non-null so you can delegate the initialization to a `lazy` function:
```kotlin
private var mDrawerToggle: ActionBarDrawerToggle by lazy { setupDrawer }

 /**
   * Set up the navigation drawer
   */
  private fun setupDrawer() = object : ActionBarDrawerToggle(this, drawerLayout, R.string.drawer_open, R.string.drawer_close) {

    override fun isDrawerIndicatorEnabled() = true

    /** Called when a drawer has settled in a completely open state.  */
    override fun onDrawerOpened(drawerView: View) {
      super.onDrawerOpened(drawerView)
      supportActionBar?.title = title
      invalidateOptionsMenu() // creates call to onPrepareOptionsMenu()
    }

    /** Called when a drawer has settled in a completely closed state.  */
    override fun onDrawerClosed(view: View) {
      super.onDrawerClosed(view)
      invalidateOptionsMenu() // creates call to onPrepareOptionsMenu()
    }
  }
```

Then in your `onCreate` you can use
```kotlin
// lazy will create the toggle when it is used here
// note it's non-null
drawerLayout.addDrawerListener(mDrawerToggle)
```